modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/helpers/SearchRequestMap.java

```diff
@@ -0,0 +1,413 @@
+        source = searchRequest.source();
+    }
+
+    /**
+     * Retrieves the number of properties in the SearchSourceBuilder.
+     *
+     * @return The number of properties in the SearchSourceBuilder.
+     * @throws UnsupportedOperationException always, as the method is not supported.
+     */
+    @Override
+    public int size() {
+        throw new UnsupportedOperationException("size" + UNSUPPORTED_OP_ERR);
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder is empty.
+     *
+     * @return {@code true} if the SearchSourceBuilder is empty, {@code false} otherwise.
+     */
+    @Override
+    public boolean isEmpty() {
+        return source == null;
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder contains the specified property.
+     *
+     * @param key The property to check for.
+     * @return {@code true} if the SearchSourceBuilder contains the specified property, {@code false} otherwise.
+     */
+    @Override
+    public boolean containsKey(Object key) {
+        return get(key) != null;
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder contains the specified value.
+     *
+     * @param value The value to check for.
+     * @return {@code true} if the SearchSourceBuilder contains the specified value, {@code false} otherwise.
+     * @throws UnsupportedOperationException always, as the method is not supported.
+     */
+    @Override
+    public boolean containsValue(Object value) {
+        throw new UnsupportedOperationException("containsValue" + UNSUPPORTED_OP_ERR);
+    }
+
+    /**
+     * Retrieves the value associated with the specified property from the SearchSourceBuilder.
+     *
+     * @param key The SearchSourceBuilder property whose value is to be retrieved.
+     * @return The value associated with the specified property.
+     * @throws IllegalArgumentException if the property is not a String.
+     * @throws SearchRequestMapProcessingException if the property is not supported.
+     */
+    @Override
+    public Object get(Object key) {
+        if (!(key instanceof String)) {
+            throw new IllegalArgumentException("key must be a String");
+        }
+        // This is the explicit implementation of fetch value from source
+        switch ((String) key) {
+            case "from":
+                return source.from();
+            case "size":
+                return source.size();
+            case "explain":
+                return source.explain();
+            case "version":
+                return source.version();
+            case "seq_no_primary_term":
+                return source.seqNoAndPrimaryTerm();
+            case "track_scores":
+                return source.trackScores();
+            case "track_total_hits":
+                return source.trackTotalHitsUpTo();
+            case "min_score":
+                return source.minScore();
+            case "timeout":
+                return source.timeout();
```

Comment:
How does `timeout` behave in practice? 

Since it's of type `TimeValue`, how does a script access the value? Does the script have access to e.g. the `getSeconds()` method?

Reply:
> I'm concerned about all of the getters that return non-primitive types.
> 
> Without a yaml test showing how they might be used in a script, I don't know how it would look to interact with those source properties.

It's a good point - we shouldn't include opensearch-specific classes as dependencies of painless syntax. So there are at least two routes to go on the read path: 1. return their primitive representations 2. does not support reading neither. It's worth for community discussion via an RFC and I removed all non-primitive types support in this PR.